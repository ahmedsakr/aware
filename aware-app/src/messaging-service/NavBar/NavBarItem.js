import React, {Component} from 'react';
import './NavBar.css'
class NavBarLink extends Component {
  render() {
    const {className, icon, name} = this.props;

    return (
      <a href="/" class={className + " navigation-menu-link"}>
        <span class={icon} aria-hidden="true"></span>
        <p>{name}</p>
      </a>
    );
  }
}

export default NavBarLink;

class NavBarLogo extends Component {
  render() {
    const {className} = this.props;

    return (
      <div class={className + " navigation-menu-logo"}>
        <p>AWARE</p>
      </div>
    );
  }
}

class NavBarTitle extends Component {
  render() {

    const {className, title} = this.props;

    return (
      <div class={className + " navigation-menu-title"}>
        <p>{title}</p>
      </div>
    );
  }
}

export {
  NavBarLogo,
  NavBarTitle
}