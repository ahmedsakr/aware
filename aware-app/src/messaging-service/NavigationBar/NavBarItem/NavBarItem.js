import React, {Component} from 'react';
import '../NavBar/NavBar.css';

class NavBarItem extends Component {
  render() {
    const {src, name} = this.props;

    return (
      <div onClick={() => { this.props.selectRoom(this.props.room)}} class="chat-navigation-item">
          <img src={process.env.PUBLIC_URL + src} alt={name}/>
          <p>{name.toUpperCase()}</p>
      </div>
    );
  }
}

export default NavBarItem;
