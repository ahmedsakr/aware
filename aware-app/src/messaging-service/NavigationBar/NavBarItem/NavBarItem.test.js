import React from 'react';
import { shallow } from 'enzyme';

import NavBarItem from './NavBarItem';

describe('NavBarItem', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<NavBarItem src="/icons8-monitor-64.png" name="sysc 2100" debug />);
  
    expect(component).toMatchSnapshot();
  });
});