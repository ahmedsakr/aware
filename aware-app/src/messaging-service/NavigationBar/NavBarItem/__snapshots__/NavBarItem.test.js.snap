// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarItem should render correctly in "debug" mode 1`] = `
<a
  class="chat-navigation-item"
  href="/"
>
  <img
    alt="sysc 2100"
    src="/icons8-monitor-64.png"
  />
  <p>
    SYSC 2100
  </p>
</a>
`;
